torch/nn/attention/_flex_attention.py

```diff
@@ -39,11 +39,69 @@ def _identity(
     return score
 
 
+class _BlockSparseMask:
+    kv_num_blocks: torch.Tensor
+    kv_indices: torch.Tensor
+    q_num_blocks: torch.Tensor
+    q_indices: torch.Tensor
+
+    def __init__(
+        self,
+        kv_num_blocks,
+        kv_indices,
+        q_num_blocks,
+        q_indices,
+    ):
+        self.kv_num_blocks = kv_num_blocks
+        self.kv_indices = kv_indices
+        self.q_num_blocks = q_num_blocks
+        self.q_indices = q_indices
+
+
+def _create_block_sparse_mask(
+    mask: torch.Tensor,
```

Comment:
TODO: making mask tensor 4D, currently I'm using 2D mask for debugging more simple.

Reply:
Done